ports/esp32/esp32_partition.c

```diff
@@ -214,6 +214,12 @@ STATIC mp_obj_t esp32_partition_get_next_update(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp32_partition_get_next_update_obj, esp32_partition_get_next_update);
 
+STATIC mp_obj_t esp32_partition_boot_ok() {
+    check_esp_err(esp_ota_mark_app_valid_cancel_rollback());
```

Comment:
The other functions here (set_boot, get_next_update) were named after the IDF functions, so this here could be `mark_app_valid_cancel_rollback()` to match, and make it explicit what is going on.  Also probably worth adding `mark_app_invalid_rollback_and_reboot`() as well, in case there was a problem.

Furthermore, these functions don't take a partition so are better suited to being `staticmethod`s (see eg `VfsFat.mkfs` for an example how to do this).

Are these functions available even with `CONFIG_APP_ROLLBACK_ENABLE` disabled?

Reply:
OK, I'll rename and change to classmethod.

I left mark_app_invalid_rollback_and_reboot out 'cause I don't see what it adds from a functionality point of view. If you just call reset you get the same effect (unless it does something if you call it after previously calling mark_app_valid_cancel_rollback, which is not a situation that is documented). There are more functions in esp-idf around all this, but it seems like a waste of time and space to implement them all.

I'll check what happens with CONFIG_APP_ROLLBACK_ENABLE disabled.

Reply:
> I left mark_app_invalid_rollback_and_reboot out 'cause I don't see what it adds from a functionality point of view. If you just call reset you get the same effect

Ok, makes sense to leave it out then.

> I'll check what happens with CONFIG_APP_ROLLBACK_ENABLE disabled.

The fact that this PR built ok means it's probably ok to have these functions with that option disabled.